server/app/mutations/domain_authorizations/authorize.rb

```diff
@@ -23,11 +23,16 @@ def validate
       unless grid.grid_secrets.find_by(name: LE_PRIVATE_KEY)
         add_error(:le_registration, :missing, "Let's Encrypt registration missing")
       end
-      if self.authorization_type == 'tls-sni-01'
-        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type") unless self.linked_service
+      if self.authorization_type == 'tls-sni-01' && self.linked_service.nil?
+        add_error(:linked_service, :missing, "Service link needs to be given for tls-sni-01 authorization type")
       end
       if self.linked_service
-        add_error(:linked_service, :not_found, "Linked service needs to point to existing service") unless @linked_service = resolve_service(self.grid, linked_service)
+        @lb_service = resolve_service(self.grid, linked_service)
+        if @lb_service.nil?
+          add_error(:linked_service, :not_found, "Linked service needs to point to existing service")
+        elsif @lb_service && !@lb_service.ports.include?(443)
```

Comment:
The `GridService.ports` is not an array of integers:

```
irb(main):002:0> GridService.find_by(name: 'lb').ports
=> [{"ip"=>"0.0.0.0", "protocol"=>"tcp", "node_port"=>80, "container_port"=>80}, {"ip"=>"0.0.0.0", "protocol"=>"tcp", "node_port"=>443, "container_port"=>443}]
```

Reply:
Fixed.